Main.cpp

```diff
@@ -148,8 +148,8 @@ int main (int argc, char *argv[]) {
     ws->SetPortAndWindowHandle(port, win_handle, 1); // blocking call
     VideoEncoderFF encoder(dims, FPS, ws);
 #else
-    ws->SetPortAndWindowHandle(port, win_handle, FPS); // blocking call
-    VideoEncoderMF encoder(dims, 1, ws);
+    ws->SetPortAndWindowHandle(port, win_handle, 1); // blocking call
+    VideoEncoderMF encoder(dims, FPS, ws);
```

Comment:
Please note that I'm deliberately lying to Media Foundation here by pretending that the frame-rate is 1FPS. This is based on https://stackoverflow.com/questions/49429954/mfcreatefmpeg4mediasink-does-not-generate-mse-compatible-mp4/55853057#55853057 and reduces the encoding latency to just 1 frame.

I fear that the proposed change would lead to increased encoding latency.

Reply:
Yes, I understand. With this 'lying' quality is bad. Changing encoder argument to FPS and time_scale multiplier to 1 makes quality much better (with increasing latency) and adding NotifyEndOfSegment decreases latency (to <50ms)

Reply:
Agree that it would be better to not lie about the actual framerate. However, before I can merge the PR this I need you to confirm that the encoding latency is still 1 frame?

I don't really care about the observed latency in ms, since that is a number that depends on a number of factors and will wary between systems. The important property is the encoding latency in number of frames.

Reply:
Ok, how did you check 1 frame latency?

Reply:
I just added breakpoints in `VideoEncoderMF::WriteFrameEnd` and `MP4FragmentEditor::EditStream` (with condition `size > 1000` to filter out non-image metadata).

You can then manually count how many times `WriteFrameEnd` is called before the corresponding MPEG frame gets transmitted.

Reply:
I've checked... 
With your source WriteFrameEnd is called 3-4 times before EditStream (with size > 1000),
 then (repeatedly) (1 or 2 WriteFrameEnd -> 1 or 2 EditStream)
With NotifyEndOfSegment - 2 calls to WriteFrameEnd -> first call to EditStream (with size > 1000).  Then (1 W -> 1ES)

Reply:
> I've checked...
> With your source WriteFrameEnd is called 3-4 times then first call to EditStream (with size > 1000),
> then (repeatedly) (1 or 2 WriteFrameEnd -> 1 or 2 EditStream)
> With NotifyEndOfSegment - 2 calls to WriteFrameEnd -> first call to EditStream (with size > 1000). Then (1 W -> 1ES)

Sorry, but this does not seem correct. I'm quite sure that there's a problem in your setup if you're observing 3-4 WriteFrameEnd calls before the first EditStream.

When I'm testing on the current "master" (commit 250b5d0b109413663654f62699a67c4ea3c3ceca) then I'm consistently observing 1 frame latency in the MPEG encoder. This is the call sequence you're supposed to observe:
* WriteFrameEnd (1st frame)
* WriteFrameEnd (2nd frame)
* EditStream (1st frame)
* WriteFrameEnd (3rd frame)
* EditStream (2nd frame)
* WriteFrameEnd (4th frame)
* EditStream (3rd frame)
* WriteFrameEnd (5th frame)
* EditStream (4rd frame)
* ...

Reply:
Encoder is h264, master branch: on the beginning in most cases 3 calls to WriteFrameEnd before first EditStream, sometimes 4W then 1ES, sometimes 2W then 1ES.
Output to std::cout confirms results observed with debugger and break-points 
WriteFrameEnd () : 10000000
WriteFrameEnd () : 20000000
WriteFrameEnd () : 30000000
WriteFrameEnd () : 40000000
EditStream() : 1
WriteFrameEnd () : 50000000
EditStream() : 2
WriteFrameEnd () : 60000000
EditStream() : 3
WriteFrameEnd () : 70000000
EditStream() : 4
WriteFrameEnd () : 80000000
EditStream() : 5
WriteFrameEnd () : 90000000
EditStream() : 6
WriteFrameEnd () : 100000000
EditStream() : 7
WriteFrameEnd () : 110000000
EditStream() : 8
EditStream() : 9
WriteFrameEnd () : 120000000
EditStream() : 10
WriteFrameEnd () : 130000000
EditStream() : 11
WriteFrameEnd () : 140000000
EditStream() : 12
WriteFrameEnd () : 150000000
EditStream() : 13
WriteFrameEnd () : 160000000
EditStream() : 14
WriteFrameEnd () : 170000000
EditStream() : 15
WriteFrameEnd () : 180000000
EditStream() : 16
WriteFrameEnd () : 190000000
EditStream() : 17
WriteFrameEnd () : 200000000
EditStream() : 18
WriteFrameEnd () : 210000000
EditStream() : 19
WriteFrameEnd () : 220000000
EditStream() : 20
WriteFrameEnd () : 230000000
EditStream() : 21
WriteFrameEnd () : 240000000
EditStream() : 22
EditStream() : 23
WriteFrameEnd () : 250000000
WriteFrameEnd () : 260000000
EditStream() : 24
WriteFrameEnd () : 270000000

Strange that number of 'first' calls to WriteFrameEnd are not the same every program start

Reply:
Seems like there's then something odd with either your Windows installation or encoding back-end. Suggest that you start by commenting out the `MF_READWRITE_ENABLE_HARDWARE_TRANSFORMS` lines to disable GPU acceleration. This prevents loading of the Nvidia "NVenc" DLLs on my computer.

Please report back afterward on if you're then able to achieve 1 frame latency on master.

Reply:
Yes, looks that with disabling HW Transforms it has 1 frame latency

Reply:
Good. Then you might want to contact your GPU vendor to complain about the H264 encoder bundled with your GPU driver delivering unnecessary high encoding latency.

There's been so much back&forth discussion on this PR now with sometimes misleading statements. I therefore suggest that you create close this PR and instead create a new PR where you only refer to latencies observed when disabling `MF_READWRITE_ENABLE_HARDWARE_TRANSFORMS`, so that we can reproduce one another's results.